client/src/pages/admin/mentor-registry.tsx

```diff
@@ -140,6 +140,17 @@ function Page() {
 
   const handleModalSubmit = useCallback(
     async (values: FormData) => {
+      const originalData = modalData?.record?.preselectedCourses?.map(String).sort();
+      const updatedData = values.preselectedCourses?.map(String).sort();
+
+      const isSame =
+        originalData?.length === updatedData?.length && originalData.every((value, i) => value === updatedData[i]);
```

Comment:
2) There's a simplier way to compare these values  
```suggestion
      const isSame = JSON.stringify(originalData) === JSON.stringify(updatedData);
```